ambry-protocol/src/main/java/com.github.ambry.protocol/BlobStoreControlAdminRequest.java

```diff
@@ -35,21 +35,25 @@
    */
   public static BlobStoreControlAdminRequest readFrom(DataInputStream stream, AdminRequest adminRequest)
       throws IOException {
-    Short versionId = stream.readShort();
-    if (!versionId.equals(VERSION_V1)) {
-      throw new IllegalStateException("Unrecognized version for BlobStoreControlAdminRequest: " + versionId);
-    }
+    // read the version
+    stream.readShort();
```

Comment:
So, are we just erasing all remnants of V1 of this request? I guess thats ok since this is an admin operation that isn't called by the router in a live environment. However, you should probably add some notes that say why v1 is not present?

Do you still want to have the check that the version number matches the expected value (2) ?

Reply:
I feel like maybe we can still use V1 for this request. Current change doesn't really modify serialize/deserialize logic of `BlobStoreControlAdminRequest`, it interprets `stream.readByte()` as a new introduced enum type instead of boolean value. I think keeping using V1 makes sense here (also, we can keep checking version number in this method).

Reply:
Yep, I agree that in this circumstance it makes sense to keep using v1